middlewares/authzCourse.sql

```diff
@@ -1,3 +1,28 @@
+-- BLOCK insert_xc101_viewer
+WITH
+course_permissions_for_user AS (
+    SELECT
+        *
+    FROM
+        course_permissions AS cp
+    WHERE
+        cp.user_id = $user_id
+),
+example_course AS (
+    SELECT * FROM pl_courses WHERE (options->'isExampleCourse')::boolean IS TRUE
+)
+INSERT INTO course_permissions (user_id, course_id, course_role)
+    SELECT
+        cp.user_id, xc.id, 'Viewer'
+    FROM
+        course_permissions_for_user AS cp
+        JOIN example_course AS xc ON (xc.id != cp.course_id)
+    WHERE
+        xc.id = $course_id
+        AND cp.course_role IN ('Owner', 'Editor', 'Viewer')
+    LIMIT 1
+ON CONFLICT DO NOTHING
```

Comment:
This query is testing to see whether the user is already an Owner/Editor/Viewer in another course. But I think we want to unconditionally add the user as a Viewer to XC 101 if the requested course ($course_id) is in fact XC 101?

Reply:
Ack yeah, I got mixed up. That filter should only be for the home page.